libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1497,6 +1499,11 @@ string YulUtilFunctions::copyArrayToStorageFunction(ArrayType const& _fromType,
 		*_fromType.copyForLocation(_toType.location(), _toType.isPointer()) == dynamic_cast<ReferenceType const&>(_toType),
 		""
 	);
+	if (!_toType.isDynamicallySized())
+		solAssert(!_fromType.isDynamicallySized() && _fromType.length() <= _toType.length(), "");
+	if (_fromType.isDynamicallySized())
+		solAssert(_toType.isDynamicallySized(), "");
```

Comment:
The first assertion is redundant because when `_fromType` is dynamic and `_toType` is static, the first one will fail first.

Reply:
That is true. I will remove it before merging